master/usage/policy/secure-metrics.md

```diff
@@ -0,0 +1,403 @@
+     types:
+     - Ingress
+     ingress:
+     # Deny anything that tries to access the prometheus port
+     # but that doesn't match the necessary selector.
+     - action: Deny
+       protocol: TCP
+       source:
+         notSelector: calico-prometheus-access == "true"
+       destination:
+         ports:
+         - 9091
+   ```
+
+   Then, use calicoctl to apply this policy.
+
+   ```shell
+   calicoctl apply -f calico-prometheus-policy.yaml
+   ```
+
+1. Apply labels to any endpoints which should have access to the metrics.
+
+   At this point, only endpoints which have the label `calico-prometheus-access: true` can reach
+   {{site.prodname}}'s prometheus metrics endpoints on each node. To grant access, simply add this label to the
+   desired endpoints.
+
+   For example, to allow access to a Kubernetes pod you can run the following command.
+
+   ```shell
+   kubectl label pod my-prometheus-pod calico-prometheus-access=true
+   ```
+
+   If you would like to grant access to a specific IP address in your network, you
+   can create a [global network set](../../reference/calicoctl/resources/globalnetworkset) using calicoctl.
+
+   For example, creating the following network set would grant access to a host with IP 172.15.0.101.
+
+   ```yaml
+   apiVersion: projectcalico.org/v3
+   kind: GlobalNetworkSet
+   metadata:
+     name: calico-prometheus-set
+     labels:
+       calico-prometheus-access: "true"
+   spec:
+     nets:
+     - 172.15.0.101/32
+   ```
+
+### Extended procedure for when Typha is enabled
+
+If your {{site.prodname}} installation uses the Kubernetes API datastore and has greater than 50 nodes, it is likely
+that you have installed Typha. This section shows how to use an additional network policy to secure the Typha
+prometheus endpoints.
+
+After following the steps above, create a file named `typha-prometheus-policy.yaml` with the following contents.
+
+```yaml
+# Allow traffic to prometheus only from sources which are
+# labeled as such, but don't impact any other traffic.
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: restrict-calico-node-prometheus
+spec:
+  # Select all {{site.prodname}} nodes.
+  selector: running-calico == "true"
+  order: 500
+  types:
+  - Ingress
+  ingress:
+  # Deny anything that tries to access the prometheus port
+  # but that doesn't match the necessary selector.
+  - action: Deny
+    protocol: TCP
+    source:
+      notSelector: calico-prometheus-access == "true"
+    destination:
+      ports:
+      - 9093
+```
+
+Then, use calicoctl to apply this policy.
+
+```shell
+calicoctl apply -f typha-prometheus-policy.yaml
+```
+
+## Using a whitelist approach
+
+This approach denies traffic to and from your hosts by default, and requires that all
+desired communication be explicitly allowed by a network policy.
+
+### Overview
+
+The basic process is as follows:
+
+1. Create a default network policy which denies traffic to and from your hosts.
```

Comment:
Strictly speaking this first step isn't needed but I think it's nice for symmetry and is probably best practice any way to create such a policy rather than relying on the default behavior.

Reply:
I think I'd rely on the default deny, then you don't need to worry about having a selector to match all host endpoints only.  (If you explain the creation of policy first above then that would bring the default deny behaviour into the story too.)